### ISSUE TITLE
File-level Type Controls reported as error

### ISSUE THREAD

**Issue by remss** [microsoft/pyright#8856]
**Describe the bug**
I'm using [File-level Type Controls](https://github.com/microsoft/pyright/blob/ce64adc17bf99286c32b1120471338f5d45d4150/docs/comments.md) to ignore some errors in a file. When I run pyright command to check it then it reports error "Pyright comments used to control file-level settings must appear on their own line".

**Code or Screenshots**

test_me.py
```python
# pyright: reportMissingParameterType=false,reportUnknownParameterType=false

def do_this(a, b):
    return a + b
```

pyright command line output is
```
test_me.py:1:4 - error: Pyright comments used to control file-level settings must appear on their own line
    Use "# pyright: ignore[<diagnostic rules>] to suppress diagnostics for a single line
1 error, 0 warnings, 0 informations
```

**VS Code extension or command-line**
pyright 1.1.373 to 1.1.378

**Comment by erictraut** [comment]
I'm not able to repro the problem with the steps above. Could you confirm that you are able to follow these steps and repro the issue?

The error that you are seeing is reported when one or more characters appears before the `#` character in the comment. Perhaps you have an extra space or some other invisible character on that line.

**Comment by remss** [comment]
Indeed the .py file was saved using "UTF-8 with BOM" encoding. Moving to simple "UTF-8 without BOM" solved the issue, there is no error reported any more.
Do you consider it as a bug? If not I think the documentation should mention it.
